server/base/src/main/java/org/apache/accumulo/server/AbstractServer.java

```diff
@@ -102,8 +106,40 @@ protected void updateIdleStatus(boolean isIdle) {
     }
   }
 
+  @Override
+  public void gracefulShutdown() {
+    log.info("Graceful shutdown initiated.");
+    // Don't interrupt the server thread, that will cause
+    // IO operations to fail as the servers are finishing
+    // their work.
+    requestShutdown();
```

Comment:
I think there needs to be some kind of security on this. Since it's coming via thrift, it should be possible to use the ServerContext with the SystemCredentials to verify that the request is coming from an admin utility with access to this cluster's instance.secret / config file.